### DISCUSSION TITLE
CheckRun must have detailsUrl

### DISCUSSION THREAD

**Discussion by AlistairB** [community/community#24716] [Apps, API and Webhooks]
<p>A GitHub <code>CheckRun</code> has a <code>detailsUrl</code> - <a href="https://docs.github.com/en/graphql/reference/objects#checkrun" rel="nofollow noopener">https://docs.github.com/en/graphql/reference/objects#checkrun</a> This can be <code>null</code> which is good for me, as for my app there is no external report. It is all self contained in the GitHub check run output. As such I do not set any <code>detailsUrl</code> when I update / create the check run.</p>
<p>However, it still displays a <code>View more details on MyApp</code> link, which defaults to my app’s homepage. This is confusing for the user, as they expect to see more details on the result, but do not.</p>
<p>Can there be a way to not display this link for cases like mine where there are no additional details to view on an external site?</p>

**Comment by francisfuzz** [comment]
<p>:wave: <a class="mention" href="/alistairb">@alistairb</a>: Thanks for sharing this observation with us! I’d love to see what you’re seeing: could you share an example of that <code>View more details on MyApp</code> copy in the form of a pull request in a public repository? Previous to reading this report, I’d expect that link to actually point to the respective check run URL within <a href="http://GitHub.com">GitHub.com</a>, though I’m interested in determining what might have changed on our end that would cause that to happen.</p>
<p>With that example on hand, I’ll review it and we’ll take the next steps! :+1:</p>

**Comment by AlistairB** [comment]
<p><a class="mention" href="/francisfuzz">@francisfuzz</a> sorry for the delay. For example if you look at this check run report <a href="https://github.com/deadpendency/deadpendency-example/pull/1/checks" rel="noopener nofollow ugc">https://github.com/deadpendency/deadpendency-example/pull/1/checks</a> at the bottom you can see <code>View more details on Deadpendency</code> at the bottom. This simply links to the <code>Homepage URL</code> specified in my app’s settings.</p>
<p>I would instead expect not setting the <code>detailsUrl</code> to not render this link at all. A user will be disappointed if they click this and no more details!</p>

**Comment by francisfuzz** [comment, marked as answer]
<aside class="quote no-group" data-username="alistairb" data-post="3" data-topic="125179">
<div class="title">
<div class="quote-controls"></div>
 alistairb:</div>
<blockquote>
<p>For example if you look at this check run report <a href="https://github.com/deadpendency/deadpendency-example/pull/1/checks">https://github.com/deadpendency/deadpendency-example/pull/1/checks </a> at the bottom you can see <code>View more details on Deadpendency</code> at the bottom. This simply links to the <code>Homepage URL</code> specified in my app’s settings.</p>
<p>I would instead expect not setting the <code>detailsUrl</code> to not render this link at all. A user will be disappointed if they click this and no more details!</p>
</blockquote>
</aside>
<p><a class="mention" href="/alistairb">@alistairb</a> - My apologies for the delay––I looked into this for you, and I have an update!</p>
<p>As you’ve noted in the OP, a check run’s <code>details_url</code> value can be <code>null</code> upon its creation. <code>details_url</code> could be updated later by making the appropriate call––via the <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/checks#update-a-check-run"><code>Update a check run</code> REST API endpoint</a> or the <a href="https://docs.github.com/en/free-pro-team@latest/graphql/reference/mutations#updatecheckrun"><code>updateCheckRun</code> mutation</a>––but it can also just be left alone (as you’ve done).</p>
<p>The current implementation of the <code>View more details on {Application Name}</code> will link to one of two values:</p>
<ul>
<li>the <code>details_url</code> value that was specified upon the check run’s creation (or whenever it was last updated)</li>
<li>the GitHub App’s URL as it was set <a href="https://docs.github.com/en/free-pro-team@latest/developers/apps/creating-a-github-app">during its creation</a> (or whenever it was last updated)</li>
</ul>
<p>For the case you’ve shared, no <code>details_url</code> was specified in either the call creating the check run nor in any calls updating it (if they were ever made). However, we automatically set the value of a check run’s <code>details_url</code> to the GitHub App’s URL.</p>
<p>As far as I know, there isn’t a way to disable that default behavior of automatically setting the <code>details_url</code>. Prior to this explanation and putting myself in your shoes––I could see how that can be confusing and it’s something that <a href="https://support.github.com/contact/feedback">our product team would love to more hear about</a> so they can consider that feedback for future iterations.</p>
<p>That said, because it’s possible to update a check run after its created, you could leverage that<br>
by setting the value of <code>details_url</code> to the <code>html_url</code> value defined on the check run. This avoids the confusion of being pointed to your application’s marketing website and effectively serves as a <a href="https://en.wikipedia.org/wiki/NOP_(code)">no-op function</a>. Here’s an example <code>curl</code> request for fetching that particular check run:</p>
<pre><code class="lang-auto">curl -v -H "Authorization: token $GITHUB_TOKEN" https://api.github.com/repos/deadpendency/deadpendency-example/check-runs/1109598375
</code></pre>
<details>
<summary>Check run output</summary>
<pre><code class="lang-auto">{
  "id": 1109598375,
  "node_id": "MDg6Q2hlY2tSdW4xMTA5NTk4Mzc1",
  "head_sha": "1ebed60a97dd42ec61c29e357271194fb35473f3",
  "external_id": "",
  "url": "https://api.github.com/repos/deadpendency/deadpendency-example/check-runs/1109598375",
  "html_url": "https://github.com/deadpendency/deadpendency-example/runs/1109598375",
  "details_url": "https://deadpendency.com",
  "status": "completed",
  "conclusion": "failure",
  "started_at": "2020-09-13T23:51:35Z",
  "completed_at": "2020-09-13T23:51:44Z",
  "output": {
    "title": "Deadpendency Check Result",
    "summary": "Unhealthy dependencies detected! 🧟",
    "text": "&lt;h2&gt;Dependency Reports&lt;/h2&gt;&lt;h3&gt;Unhealthy 🧟&lt;/h3&gt;&lt;ul&gt;&lt;li id=\"dependency-report-failure\"&gt;&lt;strong&gt;jquery-ui&lt;/strong&gt; &amp;ndash; &lt;a href=\"https://github.com/jquery/jquery-ui\" id=\"source-repo-link\" title=\"source repository\"&gt;📄&lt;/a&gt; &lt;a href=\"https://www.npmjs.com/package/jquery-ui\" id=\"package-repo-link\" title=\"registry package\"&gt;📦&lt;/a&gt;&lt;ul&gt;&lt;li id=\"failure\"&gt;❌ Last package release over 12 months ago. Last release: 2016-09-14&lt;/li&gt;&lt;/ul&gt;&lt;/li&gt;&lt;li id=\"dependency-report-failure\"&gt;&lt;strong&gt;nomnom&lt;/strong&gt; &amp;ndash; &lt;a href=\"https://github.com/harthur/nomnom\" id=\"source-repo-link\" title=\"source repository\"&gt;📄&lt;/a&gt; &lt;a href=\"https://www.npmjs.com/package/nomnom\" id=\"package-repo-link\" title=\"registry package\"&gt;📦&lt;/a&gt;&lt;ul&gt;&lt;li id=\"failure\"&gt;❌ No commits within the last 9 months.&lt;/li&gt;&lt;li id=\"failure\"&gt;❌ Package has been deprecated in the package registry. More Info: Package no longer supported. Contact support@npmjs.com for more info.&lt;/li&gt;&lt;li id=\"failure\"&gt;❌ Last package release over 12 months ago. Last release: 2014-11-07&lt;/li&gt;&lt;/ul&gt;&lt;/li&gt;&lt;li id=\"dependency-report-failure\"&gt;&lt;strong&gt;(github.com:deadpendency/an-archived-repo)&lt;/strong&gt; &amp;ndash; &lt;a href=\"https://github.com/deadpendency/an-archived-repo\" id=\"source-repo-link\" title=\"source repository\"&gt;📄&lt;/a&gt;&lt;ul&gt;&lt;li id=\"failure\"&gt;❌ Only 1 commit(s) within last year. Expected more than 4.&lt;/li&gt;&lt;li id=\"failure\"&gt;❌ Repository is archived.&lt;/li&gt;&lt;/ul&gt;&lt;/li&gt;&lt;/ul&gt;&lt;h3&gt;Healthy With Warnings 🤒&lt;/h3&gt;&lt;ul&gt;&lt;li id=\"dependency-report-warning\"&gt;&lt;strong&gt;(github.com:deadpendency/jquery-fork)&lt;/strong&gt; &amp;ndash; &lt;a href=\"https://github.com/deadpendency/jquery-fork\" id=\"source-repo-link\" title=\"source repository\"&gt;📄&lt;/a&gt;&lt;ul&gt;&lt;li id=\"warning\"&gt;⚠ The repository is a fork.&lt;/li&gt;&lt;/ul&gt;&lt;/li&gt;&lt;/ul&gt;&lt;h3&gt;Healthy 😊&lt;/h3&gt;&lt;ul&gt;&lt;li id=\"dependency-report-pass\"&gt;@testing-library/jest-dom &amp;ndash; &lt;a href=\"https://github.com/testing-library/jest-dom\" id=\"source-repo-link\" title=\"source repository\"&gt;📄&lt;/a&gt; &lt;a href=\"https://www.npmjs.com/package/@testing-library/jest-dom\" id=\"package-repo-link\" title=\"registry package\"&gt;📦&lt;/a&gt;&lt;/li&gt;&lt;li id=\"dependency-report-pass\"&gt;@testing-library/react &amp;ndash; &lt;a href=\"https://github.com/testing-library/react-testing-library\" id=\"source-repo-link\" title=\"source repository\"&gt;📄&lt;/a&gt; &lt;a href=\"https://www.npmjs.com/package/@testing-library/react\" id=\"package-repo-link\" title=\"registry package\"&gt;📦&lt;/a&gt;&lt;/li&gt;&lt;li id=\"dependency-report-pass\"&gt;@testing-library/user-event &amp;ndash; &lt;a href=\"https://github.com/testing-library/user-event\" id=\"source-repo-link\" title=\"source repository\"&gt;📄&lt;/a&gt; &lt;a href=\"https://www.npmjs.com/package/@testing-library/user-event\" id=\"package-repo-link\" title=\"registry package\"&gt;📦&lt;/a&gt;&lt;/li&gt;&lt;li id=\"dependency-report-pass\"&gt;lodash &amp;ndash; &lt;a href=\"https://github.com/lodash/lodash\" id=\"source-repo-link\" title=\"source repository\"&gt;📄&lt;/a&gt; &lt;a href=\"https://www.npmjs.com/package/lodash\" id=\"package-repo-link\" title=\"registry package\"&gt;📦&lt;/a&gt;&lt;/li&gt;&lt;li id=\"dependency-report-pass\"&gt;react &amp;ndash; &lt;a href=\"https://github.com/facebook/react\" id=\"source-repo-link\" title=\"source repository\"&gt;📄&lt;/a&gt; &lt;a href=\"https://www.npmjs.com/package/react\" id=\"package-repo-link\" title=\"registry package\"&gt;📦&lt;/a&gt;&lt;/li&gt;&lt;li id=\"dependency-report-pass\"&gt;react-scripts &amp;ndash; &lt;a href=\"https://github.com/facebook/create-react-app\" id=\"source-repo-link\" title=\"source repository\"&gt;📄&lt;/a&gt; &lt;a href=\"https://www.npmjs.com/package/react-scripts\" id=\"package-repo-link\" title=\"registry package\"&gt;📦&lt;/a&gt;&lt;/li&gt;&lt;li id=\"dependency-report-pass\"&gt;underscore &amp;ndash; &lt;a href=\"https://github.com/jashkenas/underscore\" id=\"source-repo-link\" title=\"source repository\"&gt;📄&lt;/a&gt; &lt;a href=\"https://www.npmjs.com/package/underscore\" id=\"package-repo-link\" title=\"registry package\"&gt;📦&lt;/a&gt;&lt;/li&gt;&lt;/ul&gt;&lt;h3&gt;Ignored&lt;/h3&gt;&lt;ul&gt;&lt;li id=\"dependency-report-ignore\"&gt;react-dom&lt;/li&gt;&lt;/ul&gt;&lt;p id=\"deadpendency-version\"&gt;Deadpendency Version: 0.1.1&lt;/p&gt;",
    "annotations_count": 0,
    "annotations_url": "https://api.github.com/repos/deadpendency/deadpendency-example/check-runs/1109598375/annotations"
  },
  "name": "Deadpendency Check",
  "check_suite": {
    "id": 1174935720
  },
  "app": {
    "id": 51614,
    "slug": "deadpendency",
    "node_id": "MDM6QXBwNTE2MTQ=",
    "owner": {
      "login": "deadpendency",
      "id": 60385638,
      "node_id": "MDEyOk9yZ2FuaXphdGlvbjYwMzg1NjM4",
      "avatar_url": "https://avatars2.githubusercontent.com/u/60385638?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/deadpendency",
      "html_url": "https://github.com/deadpendency",
      "followers_url": "https://api.github.com/users/deadpendency/followers",
      "following_url": "https://api.github.com/users/deadpendency/following{/other_user}",
      "gists_url": "https://api.github.com/users/deadpendency/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/deadpendency/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/deadpendency/subscriptions",
      "organizations_url": "https://api.github.com/users/deadpendency/orgs",
      "repos_url": "https://api.github.com/users/deadpendency/repos",
      "events_url": "https://api.github.com/users/deadpendency/events{/privacy}",
      "received_events_url": "https://api.github.com/users/deadpendency/received_events",
      "type": "Organization",
      "site_admin": false
    },
    "name": "Deadpendency",
    "description": "Deadpendency assesses your repository dependencies from a project health standpoint. It produces a report identifying warnings and errors which suggest that a dependency may not be actively maintained or have other issues. For example, it may be deprecated in the package registry.\r\n\r\nThe deadpendency check runs for each push to a repository producing a check run report. Unhealthy dependencies fail the check. [See an example report](https://github.com/deadpendency/deadpendency-example/pull/1/checks).\r\n\r\nSee [deadpendency.com](https://deadpendency.com) for more information.",
    "external_url": "https://deadpendency.com",
    "html_url": "https://github.com/apps/deadpendency",
    "created_at": "2020-01-19T07:05:42Z",
    "updated_at": "2020-09-13T23:55:50Z",
    "permissions": {
      "checks": "write",
      "contents": "read",
      "metadata": "read"
    },
    "events": [
      "check_suite"
    ]
  },
  "pull_requests": [
    {
      "url": "https://api.github.com/repos/deadpendency/deadpendency-example/pulls/1",
      "id": 449277713,
      "number": 1,
      "head": {
        "ref": "add-dependency",
        "sha": "1ebed60a97dd42ec61c29e357271194fb35473f3",
        "repo": {
          "id": 279760159,
          "url": "https://api.github.com/repos/deadpendency/deadpendency-example",
          "name": "deadpendency-example"
        }
      },
      "base": {
        "ref": "master",
        "sha": "f3e2b5456963dcac90d9c9b7fcbbe2bf07bfbeeb",
        "repo": {
          "id": 279760159,
          "url": "https://api.github.com/repos/deadpendency/deadpendency-example",
          "name": "deadpendency-example"
        }
      }
    }
  ]
}
</code></pre>
</details>
<p>Walking through all of that, my <em>personal opinion</em> (as I know our development team might see it differently :wink:) is to provide a way for integrators to disable that particular view component programmatically or override the <code>details_url</code> with an indicator for doing so.</p>
<p>I hope this context helps!</p>

**Comment by AlistairB** [comment]
<p>Thanks for the update. I will send feedback by the feedback form. Cheers.</p>
